test/trains.js

```diff
@@ -0,0 +1,64 @@
+
+const chai = require( 'chai' )
+const chaiAsPromised = require( 'chai-as-promised' )
+const expect = chai.expect
+const assert = chai.assert
+const db = require( '../db/config' ).db
+const Train = require( '../db/commands/train' )
+
+chai.use( chaiAsPromised )
+
+describe('Train', function() {
+  let firstTrain, secondTrain
+
+  beforeEach( function() {
+    return Promise.all([
+      db.query( 'TRUNCATE trains' ),
+      firstTrain = new Train( { trainNumber: 1, currentStation: 'Downtown', nextStation: 'Elm Street' } ),
+      secondTrain = new Train( { trainNumber: 2, currentStation: 'Annex', nextStation: '10th Ave' } )
+    ])
+    .then( () => firstTrain.save() )
+    .then( () => secondTrain.save() )
+  })
+
+  it('should be a function', function() {
+    expect( Train ).to.be.a( 'function' )
```

Comment:
if this was not true the beforeEach would fail so this test should probably be independant of the beforeEach you have above?